actors/miner/tests/exported_getters.rs

```diff
@@ -0,0 +1,150 @@
+                Method::IsControllingAddressExported as u64,
+                &serialize(&IsControllingAddressParam { address: *control }, "serializing control")
+                    .unwrap(),
+            )
+            .unwrap()
+            .deserialize()
+            .unwrap();
+        assert!(is_control_ret.is_controlling);
+
+        rt.verify();
+    }
+
+    // check that a non-controlling address doesn't return true
+
+    rt.expect_validate_caller_any();
+    let is_control_ret: IsControllingAddressReturn = rt
+        .call::<Actor>(
+            Method::IsControllingAddressExported as u64,
+            &serialize(
+                &IsControllingAddressParam { address: INIT_ACTOR_ADDR },
+                "serializing control",
+            )
+            .unwrap(),
+        )
+        .unwrap()
+        .deserialize()
+        .unwrap();
+    assert!(!is_control_ret.is_controlling);
+
+    rt.verify();
+
+    // sector size is good
+    rt.expect_validate_caller_any();
+    let sector_size_ret: GetSectorSizeReturn = rt
+        .call::<Actor>(Method::GetSectorSizeExported as u64, &RawBytes::default())
+        .unwrap()
+        .deserialize()
+        .unwrap();
+
+    rt.verify();
+
+    assert_eq!(h.sector_size, sector_size_ret.sector_size);
+
+    h.check_state(&rt);
+}
+
+#[test]
+fn collateral_getters() {
+    let h = ActorHarness::new(PERIOD_OFFSET);
+    let mut rt = h.new_runtime();
+    rt.balance.replace(BIG_BALANCE.clone());
+
+    let precommit_epoch = PERIOD_OFFSET + 1;
+    rt.set_epoch(precommit_epoch);
+
+    h.construct_and_verify(&mut rt);
+    let dl_info = h.deadline(&rt);
+
+    // Precommit a sector
+    // Use the max sector number to make sure everything works.
+    let sector_no = MAX_SECTOR_NUMBER;
+    let prove_commit_epoch = precommit_epoch + rt.policy.pre_commit_challenge_delay + 1;
+    let expiration =
+        dl_info.period_end() + DEFAULT_SECTOR_EXPIRATION * rt.policy.wpost_proving_period; // something on deadline boundary but > 180 days
+
+    let precommit_params =
+        h.make_pre_commit_params(sector_no, precommit_epoch - 1, expiration, vec![]);
+    let precommit =
+        h.pre_commit_sector_and_get(&mut rt, precommit_params, PreCommitConfig::empty(), true);
+
+    // run prove commit logic
+    rt.set_epoch(prove_commit_epoch);
+    let actor_balance = TokenAmount::from_whole(1000);
+    rt.balance.replace(actor_balance.clone());
+    let pcc = ProveCommitConfig::empty();
+
+    let sector = h
+        .prove_commit_sector_and_confirm(
+            &mut rt,
+            &precommit,
+            h.make_prove_commit_params(sector_no),
+            pcc,
+        )
+        .unwrap();
+
+    // query available balance
+
+    rt.expect_validate_caller_any();
+    let available_balance_ret: GetAvailableBalanceReturn = rt
+        .call::<Actor>(Method::GetAvailableBalanceExported as u64, &RawBytes::default())
+        .unwrap()
+        .deserialize()
+        .unwrap();
```

Comment:
I think it would be reasonable, perhaps even better, to fold this into existing tests. Some of them must already check balance components etc. Add this to the harness and then call it from sector tests (possibly replacing some other balance checks).

Reply:
Yeah, I like what you did with the market test.